pkg/clusteragent/admission/common/label_selectors.go

```diff
@@ -13,64 +13,76 @@ import (
-		len(pkgconfigsetup.Datadog().GetStringSlice("apm_config.instrumentation.enabled_namespaces")) > 0 {
-		// Accept all, ignore pods if they're explicitly filtered-out
-		labelSelector = metav1.LabelSelector{
-			MatchExpressions: []metav1.LabelSelectorRequirement{
-				{
-					Key:      EnabledLabelKey,
-					Operator: metav1.LabelSelectorOpNotIn,
-					Values:   []string{"false"},
-				},
-			},
-		}
+	// IncludeNamespaces lists namespaces to explicitly include (mutually exclusive with ExcludeNamespaces)
+	// If set, only these namespaces will be targeted
+	IncludeNamespaces []string
+}
+
+// DefaultLabelSelectors returns namespace and object selectors for admission webhooks.
+//
+// If useNamespaceSelector is true, only a namespace selector will be used for matching. This is necessary
+// for older Kubernetes versions that do not support object selectors.
+func DefaultLabelSelectors(useNamespaceSelector bool, config LabelSelectorsConfig) (nsSelector *metav1.LabelSelector, objSelector *metav1.LabelSelector) {
+	nsSelector = &metav1.LabelSelector{}
+	if useNamespaceSelector {
+		applyAdmissionEnabledSelectors(nsSelector)
 	} else {
-		// Ignore all, accept pods if they're explicitly allowed
-		labelSelector = metav1.LabelSelector{
-			MatchLabels: map[string]string{
-				EnabledLabelKey: "true",
-			},
-		}
+		objSelector = &metav1.LabelSelector{}
+		applyAdmissionEnabledSelectors(objSelector)
 	}
 
+	applySelectorConfig(nsSelector, config)
+
 	if pkgconfigsetup.Datadog().GetBool("admission_controller.add_aks_selectors") {
-		return aksSelectors(useNamespaceSelector, labelSelector)
+		// Azure AKS adds the namespace selector even in Kubernetes versions that
+		// support object selectors, so we need to add it to avoid conflicts.
+		nsSelector.MatchExpressions = append(
+			nsSelector.MatchExpressions,
+			azureAKSLabelSelectorRequirement()...,
+		)
 	}
+	return nsSelector, objSelector
+}
 
-	if useNamespaceSelector {
-		return &labelSelector, nil
+func applySelectorConfig(nsSelector *metav1.LabelSelector, config LabelSelectorsConfig) {
+	if len(config.IncludeNamespaces) > 0 {
+		nsSelector.MatchExpressions = append(nsSelector.MatchExpressions, metav1.LabelSelectorRequirement{
+			Key:      NamespaceLabelKey,
+			Operator: metav1.LabelSelectorOpIn,
+			Values:   config.IncludeNamespaces,
+		})
+	} else if len(config.ExcludeNamespaces) > 0 {
+		nsSelector.MatchExpressions = append(nsSelector.MatchExpressions, metav1.LabelSelectorRequirement{
+			Key:      NamespaceLabelKey,
+			Operator: metav1.LabelSelectorOpNotIn,
+			Values:   config.ExcludeNamespaces,
+		})
 	}
-
-	return nil, &labelSelector
 }
 
-// aksSelectors takes a label selector and builds a namespace and object
-// selector adapted for AKS. AKS adds automatically some selector requirements
-// if we don't, so we need to add them to avoid conflicts when updating the
-// webhook.
-//
-// Ref: https://docs.microsoft.com/en-us/azure/aks/faq#can-i-use-admission-controller-webhooks-on-aks
```

Comment:
It might be good to keep this link at least, otherwise it could be confusing to understand why we’re doing this